include/flox/core/exceptions.hh

```diff
@@ -32,14 +35,37 @@ enum error_category {
 /** Typed exception wrapper used for misc errors. */
 class FloxException : public std::exception {
   private:
-    std::string msg;
+    // corresponds to an error_category, in this case FLOX_EXCEPTION
+    static constexpr std::string_view categoryMsg = "error encountered running pkgdb";
```

Comment:
Question for my understanding: why is this a `string_view` and not a `string`?

Reply:
A string view is an optimized "read only reference to a string".

In your head you can essentially read it as `const std::string &` ( with flames painted on it that makes it a bit faster and fewer compatibility headaches when used in shared libraries )

Reply:
I think I used it instead of a string because string was not constexpr